addons/stock/models/product.py

```diff
@@ -599,8 +593,10 @@ class ProductTemplate(models.Model):
     )
     @api.depends_context('company')
     def _compute_quantities(self):
-        res = self._compute_quantities_dict()
-        for template in self:
+        # disable active_test to count also archived products
+        new_self = self.with_context(active_test=False)
```

Comment:
not sure you need to replace the whole self's context?
couldn't you just adapte in _compute_quantities_dict
``` python
        Move = self.env['stock.move']
        Quant = self.env['stock.quant']

```
to
``` python
        Move = self.env['stock.move'].with_context(active_test=False)
        Quant = self.env['stock.quant'].with_context(active_test=False)
```

Reply:
Changed. We also need to adapt the context every time we get product_variant_ids from the product.